src/common/utils/ldap/ldap.go

```diff
@@ -220,6 +220,27 @@ func (session *Session) SearchUser(username string) ([]models.LdapUser, error) {
 			}
 			u.GroupDNList = groupDNList
 		}
+
+		log.Debugf("Searching for nested groups")
+		nestedGroupDNList := []string{}
+		nestedGroupFilter := createNestedGroupFilter(ldapEntry.DN)
+		result, err := session.SearchLdap(nestedGroupFilter)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, groupEntry := range result.Entries {
```

Comment:
Thanks for the PR.  I think this will be part of 1.9.

However, it looks we only need a `difference(s1 []string, s2[]string) []string` to return the elements in `result.Entries` but not in `u.GroupDNList`?  So that the nested loop can be eliminated?

Additionally, could you write UT?

Reply:
+1 on the nested loop, I'll fix that today. As for UT I'll try, I'm far from being a good enough developer for that xD (Not my line of work) So if you want to help on that it'd be appreciate and probably help me learn a bit :p

Reply:
So I'm looking at that and you can't really remove the loop, since `result.Entries` is a `[]*ldap.Entry`. Unless I'm missing something you still need to go through it and get the DN from each `ldap.Entry`. I can clean it by moving it to a single function so that the above code would be just 1 line: `groupList := difference(result.Entries, u.GroupDNList)`

Reply:
@reasonerjt  ^